### PR TITLE
Tmpctx cleanups

### PR DESCRIPTION
**Pull request by rustyrussell**

### CODE REVIEW

**Comment by rustyrussell** [comment]
True, we could do that.

File: common/io_debug.c [hunk 1]
```diff
@@ -1,17 +0,0 @@
-#include <ccan/err/err.h>
```
Comment (ZmnSCPxj) [inline common/io_debug.c]: This was removed but the `lightningd/Makefile` `channeld/Makefile` still seems to use it?

Comment (rustyrussell) [reply common/io_debug.c]: Thanks!  Fixed and rebased...

**Comment by ZmnSCPxj** [comment]
ACK 4bde63993e6ee94b44134943866672fdc07e2351

Looks fine to me, although, there is a valgrind warn in a CI run, but I feel it could be rerun?

**Comment by ZmnSCPxj** [comment]
ACK 3c3fcf0bc0879c968af0d729c08876c82ffca91c

**Pull request merged**